lib/perl/Genome/Qc/Tool.pm

```diff
@@ -65,8 +65,12 @@ sub sample {
     my $self = shift;
 
     my @samples = uniq map {$_->sample} $self->alignment_result->instrument_data;
-    die "More than one sample" if scalar(@samples) > 1;
-    return $samples[0];
+    if (scalar(@samples) > 1) {
+        die $self->error_message("More than one sample: %s", join(', ', map {$_->name} @samples));
```

Comment:
Would it be useful for this error message to include the Alignment Result ID as well?